rust-enjoyer/src/main.rs

```diff
@@ -0,0 +1,124 @@
+#[cfg(target_feature = "avx2")]
+use rust_enjoyer::sponges_avx::SpongesAvx;
+
+use rust_enjoyer::*;
+
+use rayon::prelude::*;
+use std::env;
+use std::process;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+
+fn main() {
+    #[cfg(target_feature = "avx2")]
+    println!("AVX2 enabled");
+    #[cfg(not(target_feature = "avx2"))]
+    println!("AVX2 disabled");
+
+    let args: Vec<String> = env::args().collect();
+    if args.len() < 4 {
+        println!("Usage: <function name> <function params> <target selector>");
+        process::exit(-1);
+    }
+
+    // remove any leading 0x
+    let selector = args[3].to_lowercase();
+    let selector = selector.trim_start_matches("0x");
+    let selector = u32::from_str_radix(selector, 16)
+        .expect("Invalid number")
+        .to_be();
+    let function_name = SmallString::new(&args[1]);
+    let function_params = SmallString::new(&args[2]);
+
+    if function_name.length + function_params.length >= 115 {
+        println!("Total length of <function name> and <function params> must be under 115 bytes.");
+        process::exit(-1);
+    }
+
+    if std::mem::size_of::<u64>() != 8 {
+        println!("Incompatible architecture");
+        println!("u64: {}", std::mem::size_of::<u64>());
+        process::exit(-1);
+    }
+
+    println!("Function name: {}", args[1]);
+    println!("Function params: {}", args[2]);
+    println!("Target selector: {selector:X?}",);
```

Comment:
```suggestion
    println!("Target selector: {:#10x}", selector.to_be());
```

Reply:
`selector.to_be()` is done on line 29 above, and is also used for the search

is this correct?

Reply:
yes